doc/api/packages.md

```diff
@@ -612,17 +612,23 @@ The following condition definitions are currently endorsed by Node.js:
 * `"browser"` - any environment which implements a standard subset of global
   browser APIs available from JavaScript in web browsers, including the DOM
   APIs.
+* `"deno"` - indicates a variation for the Deno platform.
 * `"development"` - can be used to define a development-only environment
   entry point. _Must always be mutually exclusive with `"production"`._
 * `"production"` - can be used to define a production environment entry
   point. _Must always be mutually exclusive with `"development"`._
+* `"source"` - indicates the original .js source file without minification
+  or bundling optimizations, useful for development and debugging workflows.
```

Comment:
Is there an expectation that this is runnable? Like could this point to TypeScript files, or JavaScript that uses unsupported/experimental features like decorators?

Reply:
Personally, the sourcemaps associated with a file are always going to be how you locate "original sources" - they contain strictly more information than a simple export map field can express. I don't know what tooling uses such a condition.

Reply:
I hope it can point to TypeScript files, which can be made runnable using ESM loaders or require extensions.

Reply:
There is no reason that `"exports"` can't point to `.ts` files if targetting a platform that supports TypeScript (eg when using the `"deno"` condition). But the expectation for the `"source"` condition would typically be `.js` so the sources after running tsc but before bundling. Source maps are good and well, but if you've got a monorepo with `tsc` running, and the bundler optimization as a separate step it's a very useful workflow and a fast monorepo dev process. Many users don't yet know about `--enable-source-maps` and they do still have debuggability issues anyway. I can appreciate if deferring this condition makes sense for now though.

Reply:
I feel like this condition is very specific to your use case / expectations, so it probably makes sense to discuss it separately from the other ones.

Reply:
I'd agree, the rest seem to match up with using the whole loader pipeline for something, but this is explicitly asking loaders not to perform part of the pipeline if they would normally do so.

Reply:
As a naïve user I would expect this to be whatever the original sources were, like what `"module"` has traditionally pointed to. If this is supposed to point to runnable-but-unminified JavaScript, perhaps a better name would suggest that.

Reply:
Isn't that _generally_ what the `development` condition would be used for?

Reply:
@weswigham no, "development" is supposed to be a replacement for the `process.env.NODE_ENV === 'production'` condition for providing different behaviour between development and production environments for library consumers like React does.

If this is unclear perhaps we can clarify the current definition further.

@bmeck @targos agreed `"source"` might need some more thought for now, will remove it.

Reply:
@weswigham this is a good point though - we could indicate that the "production" condition can be used for minification to try and encourage such a pattern. The question is really then - do we want to encourage such a pattern? My concern with it is that most applications will always have a dependency that is unminified, so the minification status of individual dependencies is largely irrelevant if just one implies needing to reminify the build anyway.

Reply:
I have a slight preference that if the expectation is to be in production, we don't define an explicit condition for it and instead people should use guards against production like `"development"`. Having both worries me a little bit since it complicates the number of values in order to represent what is possible to do already (if a little more verbose for some trees)

Reply:
I tend to feel the same way and that was why it's a `"source"` condition and not an `"optimized"` condition. I still feel it is independent to `"development"` though somewhat, in that a development build can still be optimized.

Point taken on the `"production"` condition, it could be worth considering deprecating this condition in due course even.